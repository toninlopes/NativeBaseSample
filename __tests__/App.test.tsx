import 'react-native';
import React from 'react';
import {describe, it, expect} from '@jest/globals';
import {render, screen} from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import { Button } from 'native-base';
import { Text } from 'react-native';
import { NativeBaseProvider } from 'native-base';
import { Metrics, SafeAreaProvider } from 'react-native-safe-area-context';

const MOCK_INITIAL_METRICS: Metrics = {
  frame: { width: 320, height: 640, x: 0, y: 0 },
  insets: { left: 0, right: 0, bottom: 0, top: 0 },
};

const Wrapper = ({children}: {children: React.ReactNode}) =>
  <SafeAreaProvider initialMetrics={MOCK_INITIAL_METRICS}>
    <NativeBaseProvider>
      {children}
    </NativeBaseProvider>
  </SafeAreaProvider>;


describe('Snapshot Testing', () => {

  it('Render NativeBase Button with @testing-library/react-native', () => {
    render(
      <Wrapper>
        <Button>Test</Button>
      </Wrapper>
    );
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('Render NativeBase Button with react-test-renderer', () => {
    const component = renderer.create(
      <Wrapper>
        <Button>Test</Button>
      </Wrapper>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('Render React Native Text with @testing-library/react-native', () => {
    render(
      <Text>Test</Text>
    );
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('Render React Native Text with react-test-renderer', () => {
    const component = renderer.create(
      <Text>Test</Text>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });

});
