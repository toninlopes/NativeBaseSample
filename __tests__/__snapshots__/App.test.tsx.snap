// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing Render NativeBase Button with @testing-library/react-native 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      colorScheme="primary"
      dataSet={{}}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#0891b2",
          "borderRadius": 4,
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingBottom": 10,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 10,
        }
      }
    >
      <View
        dataSet={{}}
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
        test={true}
      >
        <View
          dataSet={{}}
          style={{}}
        >
          <Text
            dataSet={{}}
            style={
              {
                "backgroundColor": undefined,
                "color": "#fafafa",
                "fontFamily": undefined,
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 21,
                "textDecorationLine": undefined,
              }
            }
          >
            Test
          </Text>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</RNCSafeAreaProvider>
`;

exports[`Snapshot Testing Render NativeBase Button with react-test-renderer 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      colorScheme="primary"
      dataSet={{}}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#0891b2",
          "borderRadius": 4,
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingBottom": 10,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 10,
        }
      }
    >
      <View
        dataSet={{}}
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
        test={true}
      >
        <View
          dataSet={{}}
          style={{}}
        >
          <Text
            dataSet={{}}
            style={
              {
                "backgroundColor": undefined,
                "color": "#fafafa",
                "fontFamily": undefined,
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 21,
                "textDecorationLine": undefined,
              }
            }
          >
            Test
          </Text>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</RNCSafeAreaProvider>
`;

exports[`Snapshot Testing Render React Native Text with @testing-library/react-native 1`] = `
<Text>
  Test
</Text>
`;

exports[`Snapshot Testing Render React Native Text with react-test-renderer 1`] = `
<Text>
  Test
</Text>
`;
